xcode/Subconscious/Shared/Library/CombineUtilities.swift

```diff
@@ -8,36 +8,87 @@
 import Foundation
 import Combine
 
-struct CombineUtilities {
-    /// Run a function asyncronously on the global DispatchQueue.
-    static func async<T>(
-        qos: DispatchQoS.QoSClass = .default,
-        execute: @escaping () -> T
-    ) -> AnyPublisher<T, Never> {
-        Future({ promise in
-            DispatchQueue.global(qos: qos).async(execute: {
-                let result: T = execute()
-                promise(.success(result))
-            })
-        })
-        .eraseToAnyPublisher()
+/// Create a Combine Future from an async closure that never fails.
+/// Async actions are run in a task and fulfil the future's promise.
+public extension Future where Failure == Never {
```

Comment:
New convenience extensions to `Future` allow us to easily create Combine Futures from async functions.

This gives us much more flexibility in constructing publishers from async functions where and when we need them. It also takes advantage of Swift's new concurrency primitives (Tasks, Actors, async/await) for more ergonomic and efficient concurrency.

This replaces the older `CombineUtilities.async` helper, which scheduled things on the global queue.